crates/aptos-telemetry-service/src/prometheus_push_metrics.rs

```diff
@@ -64,66 +65,61 @@ pub async fn handle_metrics_ingest(
-        .await
-        .into_iter()
-        .zip(filtered_clients.map(|(name, _)| name))
-        .map(|(res, name)| {
-            match res {
-                Ok(res) => {
-                    METRICS_INGEST_BACKEND_REQUEST_DURATION
-                        .with_label_values(&[
-                            &claims.peer_id.to_string(),
-                            name,
-                            res.status().as_str(),
-                        ])
-                        .observe(start_timer.elapsed().as_secs_f64());
-                    if res.status().is_success() {
-                        debug!("remote write to victoria metrics succeeded");
-                    } else {
-                        error!(
-                            "remote write failed to victoria_metrics for client {}: {}",
-                            name,
-                            res.error_for_status().err().unwrap()
-                        );
-                        return Err(());
-                    }
-                },
-                Err(err) => {
-                    METRICS_INGEST_BACKEND_REQUEST_DURATION
-                        .with_label_values(&[name, "Unknown"])
-                        .observe(start_timer.elapsed().as_secs_f64());
+    let post_futures = client.iter().map(|(name, client)| async {
+        let result = client
+            .post_prometheus_metrics(
+                metrics_body.clone(),
+                extra_labels.clone(),
+                encoding.clone().unwrap_or_default(),
+            )
+            .await;
+
+        match result {
+            Ok(res) => {
+                METRICS_INGEST_BACKEND_REQUEST_DURATION
+                    .with_label_values(&[&claims.peer_id.to_string(), name, res.status().as_str()])
+                    .observe(start_timer.elapsed().as_secs_f64());
+                if res.status().is_success() {
+                    debug!("remote write to victoria metrics succeeded");
+                } else {
                     error!(
-                        "error sending remote write request for client {}: {}",
-                        name, err
+                        "remote write failed to victoria_metrics for client {}: {}",
+                        name.clone(),
+                        res.error_for_status().err().unwrap()
                     );
                     return Err(());
-                },
-            }
-            Ok(())
-        });
+                }
+            },
+            Err(err) => {
+                METRICS_INGEST_BACKEND_REQUEST_DURATION
+                    .with_label_values(&[name, "Unknown"])
+                    .observe(start_timer.elapsed().as_secs_f64());
+                error!(
+                    "error sending remote write request for client {}: {}",
+                    name.clone(),
+                    err
+                );
+                return Err(());
+            },
+        }
+        Ok(())
```

Comment:
Silly question: is this needed?

Reply:
I think so, because I only count an error when all the backends return an error. If even one succeds, we consider it good for our replicator threshold.